test/Microsoft.AspNet.Razor.Test/Parser/CSharp/CSharpBlockTest.cs

```diff
@@ -16,6 +16,31 @@ namespace Microsoft.AspNet.Razor.Test.Parser.CSharp
     public class CSharpBlockTest : CsHtmlCodeParserTestBase
     {
         [Fact]
+        public void NestedCodeBlockWithMarkupSetsDotAsMarkup()
+        {
+            ParseBlockTest("if (true) { @if(false) { <div>@something.</div> } }",
```

Comment:
Maybe add another case with `ParseDocumentTest` just to validate if it behaves the same way.